lib/RPerl/Operation/Expression/Operator/String/Repeat.pm

```diff
@@ -22,16 +28,35 @@ sub ast_to_rperl__generate {
     { my string_hashref::method $RETURN_TYPE };
     ( my object $self, my string_hashref $modes) = @ARG;
     my string_hashref $rperl_source_group = { PMC => q{} };
+    my string_hashref $cpp_source_group = { CPP => q{} };
 
 #    RPerl::diag( 'in Operator::String::Repeat->ast_to_rperl__generate(), received $self = ' . "\n" . RPerl::Parser::rperl_ast__dump($self) . "\n" );
+    #    RPerl::diag( 'in Operator::String::Repeat->ast_to_rperl__generate(), received $self = ' . "\n" . RPerl::Parser::rperl_ast__dump($self) . "\n" );
+
 
     my string $self_class = ref $self;
     if ( $self_class eq 'Operator_111' ) {  # Operator -> SubExpression OP07_STRING_REPEAT SubExpression
+        $cpp_source_group->{CPP} .= NAME_CPPOPS_CPPTYPES() . '(';
         my string_hashref $rperl_source_subgroup = $self->{children}->[0]->ast_to_rperl__generate($modes);
         RPerl::Generator::source_group_append( $rperl_source_group, $rperl_source_subgroup );
+        $cpp_source_group->{CPP} .= ', ';
+        RPerl::Generator::source_group_append( $rperl_source_group, $rperl_source_subgroup );
         $rperl_source_group->{PMC} .= q{ } . $self->{children}->[1] . q{ };
         $rperl_source_subgroup = $self->{children}->[2]->ast_to_rperl__generate($modes);
+        print q{have $foo = '}, $foo, q{'}, "\n";
+        print q{have $bar = '}, $bar, q{'}, "\n";
```

Comment:
this is wrong! @wbraswell  isn't it? there should not be print statement atleast and also it is not for string repeat .
Have you testes this before putting PR?

Reply:
> this is wrong! @wbraswell isn't it? there should not be print statement atleast and also it is not for string repeat .
> Have you testes this before putting PR?

This is NOT wrong. Yes, absolutely there shouldn't be any print statements, but @wbraswell had asked me to ensure that all tests continue to pass in both the existing PERLOPS_PERLTYPES mode as well as the new CPPOPS_CPPTYPES mode. Now, for that to happen, I should necessarily use them from the `Operator07StringRepeat`, and that is exactly what I did, from the `program_00_bad_01.pl`. Anyways, it's not necessary to implement this print statement, because this is just for testing. Even if these are removed, the program would work, well, and good.
And also, it IS for string repeat. Why'd I even write something which is not related to string repeat? Check the code functionality, and then argue.
And yes, obviously I did test did. Nobody is that dumb to put something as a PR without testing it. Yes, I do believe that in the Microsoft software, it's a bit difficult to execute RPerl, but still, I did somehow.
Kindly check what you're talking about, and then raise an issue.
Hope that helps!

Reply:
> > this is wrong! @wbraswell isn't it? there should not be print statement atleast and also it is not for string repeat .
> > Have you testes this before putting PR?
> 
> This is NOT wrong. Yes, absolutely there shouldn't be any print statements, but @wbraswell had asked me to ensure that all tests continue to pass in both the existing PERLOPS_PERLTYPES mode as well as the new CPPOPS_CPPTYPES mode. Now, for that to happen, I should necessarily use them from the `Operator07StringRepeat`, and that is exactly what I did, from the `program_00_bad_01.pl`. Anyways, it's not necessary to implement this print statement, because this is just for testing. Even if these are removed, the program would work, well, and good.
> And also, it IS for string repeat. Why'd I even write something which is not related to string repeat? Check the code functionality, and then argue.
> And yes, obviously I did test did. Nobody is that dumb to put something as a PR without testing it. Yes, I do believe that in the Microsoft software, it's a bit difficult to execute RPerl, but still, I did somehow.
> Kindly check what you're talking about, and then raise an issue.
> Hope that helps!

I think print statements are for good/bad test files not for operator implementation code. As far as it is concerned for debugging purpose then you should remove it before putting merge request.I know what i am talking about and i am damn sure this code have <STDIN> which is plain wrong.

Reply:
See, to be plain, and simple, I was asked to ensure that all the tests continue to pass in both the existing PERLOPS_PERLTYPES mode as well as the new CPPOPS_CPPTYPES mode, and, I just did that. Nothing much, nothing less.
If you feel that the code is **plain wrong**, kindly go ahead, and help me regarding that. After all, @wbraswell had obviously told that this is a cooperative open source community where we help one another, and not demotivate, and insult.
Hope you understand. and let's clear the things that have gone wrong.

Reply:
Do you really think if i say that tests are failing then it means **insult** ? You have to update new things in tests file. you do not seem to be understanding. I am not demovating  (just letting you know that you forgot to write test file edits. :-)

Reply:
If test cases are failing, that definitely doesn't mean insult, but the way of expressing it, is definitely no less than an insult! (by the way you're commenting, I mean). 
That is what I'm asking you to do! If I've to update the test files, and I can't do it, then help me in doing it, rather than just commenting, and increasing the discussion length. If I can't seem to understand. then make me understand by modifying the code, it's as simple as that.